stdlib/public/Concurrency/Clock.cpp

```diff
@@ -52,14 +54,14 @@ void swift_get_time(
       QueryPerformanceFrequency(&freq);
       LARGE_INTEGER count;
       QueryPerformanceCounter(&count);
-      *seconds = count.QuadPart / freq.QuadPart;
-      if (freq.QuadPart < 1000000000) {
-        *nanoseconds = 
-            ((count.QuadPart % freq.QuadPart) * 1000000000) / freq.QuadPart;
-      } else {
-        *nanoseconds = 
-            (count.QuadPart % freq.QuadPart) * (1000000000.0 / freq.QuadPart);
-      }
+      // Divide count (number of ticks) by frequency (number of ticks per
+      // second) to get the counter in seconds. We also need to multiply the
+      // count by 1,000,000,000 to get nanosecond resolution. By multiplying
+      // first, we maintain high precision. The resulting value is the tick
+      // count in nanoseconds.
+      long long ns = (count.QuadPart * 1000000000) / freq.QuadPart;
```

Comment:
Couldn't `count.QuadPart * 1000000000` overflow? I think the previous code might have been the way it was in order to avoid that problem?

Reply:
`count.QuadPart` is a 64-bit value and the frequency is CPU-bound (clock can't go brrr if CPU doesn't go brrr.) The counter is a 64-bit value and, unless my math is way off, would only overflow if the system had years of uptime.

However, that's a bit handwavey—we could approach this problem in a couple of different ways:

1. Instead of using `QueryPerformanceCounter()`, use `QueryInterruptTimePrecise()` and get a value in units of 100ns, which will overflow after about 58,455 years, but which is slower to call than `QueryPerformanceCounter()`.
2. Use 128-bit multiplication and division to get the nanosecond count. Windows has various bits of 128-bit math API we could use here. I don't know what sort of performance overhead this API has, but it's arithmetic so I'd hope not _too_ bad.

Reply:
I used 128-bit math.